bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -593,6 +621,11 @@ private void updateStateFromUDPEvent(String zoneToUpdate, UdpMessage targetObjec
             updateState(channel, new DecimalType(volumeState));
         }
 
+        if (volumeDbState != -80f) {
```

Comment:
Float comparison with the `==` or `!=` operators is not recommended. Consider even replacing this type with something else, for example `BigDecimal` in order to use exact comparison.

Reply:
This comparison is only used to check whether the var had an update.
Does exact comparison matter here?

Reply:
The volume returned from the Yamaha device always is .0 or .5 and you are not able to set anything else.

Reply:
I'm not a Java expert, so I'm not very much into the details, but I would presume you can risk comparing with a value like -79.9999999997 which would not equal -80. Since your volume steps are exactly 0.5 dB, probably it would be safer to switch to `BigDecimal` where accuracy is guaranteed. See for example https://howtodoinjava.com/java-examples/correctly-compare-float-double/ or many other articles for this topic.

Reply:
Hmm okay, I can switch if you want me to.

I would just like to mention that the yamahareceiver binding implements volumeDB with float and does the same float comparison.

Reply:
There is of course a possibility that I'm wrong, so if you know that this comparison will always work on all platforms, I'm fine with it. :) Just wanted to share my concern when seeing this and checking articles online. Let me know what you prefer.

Reply:
> Just wanted to share my concern when seeing this and checking articles online.

So, I do not know if it works on all platforms and in every case, but float is used in the yamahareceiver binding and I did never encounter any issue with volumeDB in that binding.

> Let me know what you prefer.

I prefer the float solution as it is easier to read (no BigDecimal Constructing etc.), but I had a look at the article and I think I will implement the Threshold based comparison (chapter 2 of the article).

But currently, I am wondering if this comparison is even needed. I think I will have a look at the code and then take a decision.

Reply:
So I have tested much, it does not matter whether that comparison is 100% exact.

Reply:
@jlaur 
I would leave it as it is now.

Reply:
Sure, let's leave it. Just a small update of the README.md needed before we are ready to merge.